src/Router.php

```diff
@@ -17,14 +17,71 @@ public static function getInstance()
         return static::$instance;
     }
     
+    /**
+     * Loads the location of the routes file.
+     * 
+     * @return string
+     */
+    protected function getRouteFileLocation()
+    {
+        $dir = getenv('HOME') . '/.volcano-sdk';
+        
+        if (!is_dir($dir)) {
+            mkdir($dir);
+        }
+        
+        return $dir . '/routes.json';
+    }
+    
+    /**
+     * Loads Routes
+     */
     public function loadRoutes()
     {
+        $rfile = $this->getRouteFileLocation();
+        
+        if (!file_exists($rfile)) {
+            $this->loadRoutesFromService();
+            return;
+        }
+        
         $this->_routeList = json_decode(
-            file_get_contents(dirname(__FILE__) . '/../routes.json'),
+            file_get_contents($rfile),
             true
         );
     }
     
+    /**
+     * Loads the routs fromm the api endpoint and stores them locally.
+     */
+    public function loadRoutesFromService()
+    {
+        $service = new Service('endpoints');
```

Comment:
Nice, so it looks like this will automatically use the defined api_key when querying the Endpoints API?

That would be perfect since even the Endpoints API requires an api_key (and eventually we can have ACL-like endpoints on a per api_key basis. E.g., a given api_key can only access the Customer APIs, etc.).